auto-value-gson/src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonAdapterFactoryProcessor.java

```diff
@@ -121,8 +125,43 @@ public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment
       }
       String adapterName = classNameOf(type, "_");
       String qualifiedName = classNameOf(type, ".");
-      String packageName = packageNameOf(type);
-      TypeSpec typeAdapterFactory = createTypeAdapterFactory(type, elements, packageName,
+      PackageElement packageElement = packageElementOf(type);
+      String packageName = packageElement.getQualifiedName().toString();
+      List<TypeElement> applicableElements = elements.stream()
+          .filter(e -> {
+            Visibility typeVisibility = Visibility.ofElement(e);
+            switch (typeVisibility) {
+              case PRIVATE:
+                return false;
+              case DEFAULT:
```

Comment:
Won't this fail to generate implementations for public elements from different packages? I think the package should only have to match if the element is protected, no?

Reply:
Ahh, okay, that sounds good then.